components/valve/index.rst

```diff
@@ -0,0 +1,260 @@
+Valve Component
+===============
+
+.. seo::
+    :description: Instructions for setting up base valves in ESPHome.
+    :image: folder-open.svg
+
+The ``valve`` component is a generic representation of valves in ESPHome. A valve can (currently) either be *closed* or
+*open* and supports three commands: *open*, *close* and *stop*.
+
+.. figure:: images/valve-ui.png
+    :align: center
+
+.. _config-valve:
+
+Base Valve Configuration
+------------------------
+
+All valve config schemas inherit from this schema - you can set these keys for valves.
+
+.. code-block:: yaml
+
+    valve:
+      - platform: ...
+        device_class: water
+
+Configuration variables:
+
+- **name** (**Required**, string): The name for the valve.
+
+  .. note::
+
+      If you have a :ref:`friendly_name <esphome-configuration_variables>` set for your device and you want the valve
+      to use that name, you can set ``name: None``.
+
+- **device_class** (*Optional*, string): The device class for the sensor. See
+  https://www.home-assistant.io/components/valve/ for a list of available options.
+- **icon** (*Optional*, icon): Manually set the icon to use for the valve in the frontend.
+
+Advanced options:
+
+- **internal** (*Optional*, boolean): Mark this component as internal. Internal components will not be exposed to the
+  frontend (like Home Assistant). Only specifying an ``id`` without a ``name`` will implicitly set this to true.
+- **disabled_by_default** (*Optional*, boolean): If true, this entity should not be added to any client's frontend,
+  (usually Home Assistant) without the user manually enabling it (via the Home Assistant UI). Requires Home Assistant
+  2021.9 or newer. Defaults to ``false``.
+- **entity_category** (*Optional*, string): The category of the entity. See
+  https://developers.home-assistant.io/docs/core/entity/#generic-properties for a list of available options. Requires
+  Home Assistant 2021.11 or newer. Set to ``""`` to remove the default entity category.
```

Comment:
It's been a while, we don't need these now. I will make a pr to remove elsewhere
```suggestion
  (usually Home Assistant) without the user manually enabling it (via the Home Assistant UI). Defaults to ``false``.
- **entity_category** (*Optional*, string): The category of the entity. See
  https://developers.home-assistant.io/docs/core/entity/#generic-properties for a list of available options. Set to ``""`` to remove the default entity category.
```

Reply:
https://github.com/esphome/esphome-docs/pull/3788